### PR TITLE
PLT-7691 flat version get transaction per contract

### PR DESCRIPTION
**Pull request by hrajchert**
This PR adds the method `getTransactionsForContract` to the RestAPI of the `runtime-rest-client` and improves the generated documentation of related methods. To test you can build the project, modify the `poc/rest-client-flow.html` so that it uses local build and not latest release and then use the new method from the page.

Also, build the docs and make sure that the documentation makes sense.

@nhenin , @jhbertra , please note the DISCUSSION topics added in this PR

### CODE REVIEW

File: packages/runtime/client/rest/src/index.ts [hunk 1]
```diff
@@ -77,20 +78,28 @@ export interface RestAPI {
...
   ): Promise<void>;
-  //
-  //   getTransactionsForContract: Transactions.GET; // - https://docs.marlowe.iohk.io/api/get-transactions // TODO: Jamie, lets unify names
+  /**
+   * Gets a paginated list of  {@link contract.TxHeader } for a given contract.
+   * @see {@link https://docs.marlowe.iohk.io/api/get-transactions-for-contract | The backend documentation }
+   */
+  // DISCUSSION: What should this return when contractId is not found? Currently it throws an exception
+  //             with an AxiosError 404, we could return a nullable value or wrap the error into a custom
+  //             ContractNotFound error and specify it in the docs.
```
Comment (hrajchert) [inline packages/runtime/client/rest/src/index.ts:98]: cc @jhbertra @nhenin

File: packages/runtime/core/src/payout/index.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { split } from "fp-ts/lib/string.js";
 import { pipe } from "fp-ts/lib/function.js";
 import { head } from "fp-ts/lib/ReadonlyNonEmptyArray.js";
 import { TxId } from "../tx/id.js";
-import { ContractId } from "../contract/id.js";
+import { ContractIdGuard } from "../contract/id.js";
 import { AssetId, Assets } from "../asset/index.js";
 
 // QUESTION: @N.H: What is the difference between PayoutId and WithdrawalId?
```
Comment (hrajchert) [inline packages/runtime/core/src/payout/index.ts:11]: cc @nhenin

File: packages/runtime/client/rest/src/index.ts (hunk 1 above)
Comment (jhbertra) [reply packages/runtime/client/rest/src/index.ts:98]: It's fine if we throw an exception here - the caller should be sure the contract exists, not use this as a means of testing for its existence. Wrapping in a custom error is not a bad idea.

File: packages/runtime/core/src/payout/index.ts (hunk 2 above)
Comment (nhenin) [reply packages/runtime/core/src/payout/index.ts:11]: PayoutId is the id of the payout :slightly_smiling_face: meaning you’ll use it to withdraw the tokens from these payouts….
 WithdrawalId  are the id of the Tx  when you withdraw from this payout…

**Review by nhenin — APPROVED** [review]

**Pull request merged**